web3js-ext/test/role_based_key_feepayer.spec.ts

```diff
@@ -0,0 +1,141 @@
+        assert.isNotNull(signedUpdateTx.transactionHash, "Account update transaction should succeed");
+    });
+
+    // Test Case 1: Sending a normal transaction with RoleTransaction key
+    it("1. Sending a normal transaction with RoleTransaction key", async function () {
+        const valueTx = {
+            type: TxType.ValueTransfer,
+            from: generateTemporaryAccount().address,
+            to: generateTemporaryAccount().address,
+            value: toPeb("0.01"),
+            gasLimit: 100000
+        };
+
+        const signedTx = await roleTransactionAccount.signTransaction(valueTx);
+        console.log("RoleTransaction signedTx:", signedTx.transactionHash);
+        assert.isNotNull(signedTx.transactionHash, "RoleTransaction transaction should succeed");
+    });
+
+    // Test Case 2: Attempting to sign a regular transaction with RoleFeePayer key (should fail)
+    it("2. Attempting to sign a regular transaction with RoleFeePayer key (failure test)", async function () {
+        const valueTx = {
+            type: TxType.ValueTransfer,
+            from: generateTemporaryAccount().address,
+            to: generateTemporaryAccount().address,
+            value: toPeb("0.01"),
+            gasLimit: 100000
+        };
+
+        try {
+            const signedTx = await roleFeePayerAccount.signTransaction(valueTx);
+            console.log("Unexpected Success - Signed Transaction:", signedTx);
+            assert.fail("RoleFeePayer key should not sign a ValueTransfer transaction.");
+        } catch (error: any) {
+            console.log("Expected Error (RoleFeePayer):", error.message);
+            assert.isTrue(true, "Error occurred as expected due to role mismatch");
+        }
+    });
+
+    // Test Case 3: Fee Delegated transaction signed by RoleFeePayer key (should succeed)
+    it("3. Fee Delegated transaction signed by RoleTransaction and RoleFeePayer keys", async function () {
+        const feeDelegatedTx = {
+            type: TxType.FeeDelegatedValueTransfer,
+            from: generateTemporaryAccount().address,
+            to: generateTemporaryAccount().address,
+            value: toPeb("0.01"),
+            gasLimit: 100000
+        };
+
+        // Feedback2. Remove the dependencies with network connection. just check the signed result.
+        // Feedback3. signed by a user (from address,roleTransactionAccount) first
+        const signedTxBySender = await roleTransactionAccount.signTransaction(feeDelegatedTx);
+        assert.isNotNull(signedTxBySender.rawTransaction, "Transaction should be signed by RoleTransaction");
+
+        console.log("Sender signed transaction:", signedTxBySender.rawTransaction);
+
+        // Feedback3. FeePayer(FeePayerAccount) sign the tx after the user's sign.
+        try {
+            const signedTxByFeePayer = await roleFeePayerAccount.signTransaction({
+                senderRawTransaction: signedTxBySender.rawTransaction
```

Comment:
[[Feedback 3]](https://github.com/kaiachain/kaia-sdk/pull/59/files#r1843203316) It should be signed by a user (from address) first. FeePayer normally sign the tx after the user's sign.


Actually, I have two questions about this Feedback3.

1. when you said "It should be signed by a user (from address) first. FeePayer normally sign the tx after the user's sign.", Is it correct to sign with roleTransactionAccount(user who signs the transaction account) first and then append roleFeePayerAccount to the rawTransaction of that signature?

2. If there is one unexpected issue in this test, it is “Returned error: insufficient funds for transfer”. I created temporary addresses via [Feedback 1], but now that accounts don't have enough klay(kaia). Is there a good way to charge klay(kaia) to the temporary accounts...? (I can't think of a good way... T^T...)

cc. @kjeom

Reply:
@minminkikiki "A user" means another account that is used by user, not feePayer.
roleTransactionKey and roleFeePayerKey are owned by feePayer. That means all keys of each role belong to another key owner who is not the user.

Reply:
@minminkikiki If you got the new error regarding the sign of feepayer, please add the description of the error in the PR description.